ops/framework.py

```diff
@@ -235,7 +235,7 @@ def _set_name(self, emitter_type, event_kind):
                     self.event_kind,
                     emitter_type.__name__,
                     event_kind,
-                ))
+                )) from None
```

Comment:
Out of interest, why is "from None" needed here?

Reply:
Essentially to head off excessive exception trees from core Python. `__setname__` (like some builtins) has a very, uh, hardcoded exception handler because of the way it looks it up in class attributes. See [here](https://bugs.python.org/issue28214). Without this, even `exc.__cause__` requires multiple levels of nesting in the testsuite, but `from None` keeps the "user-facing" output the same as it was in the previous versions of OF.